### PR TITLE
Create Api For Gym Search 

### PR DESCRIPTION
**Pull request by AFahgan**
-  Create route /api/v1/gyms/seach?q
-  Create controller
-  Handle data Form the database

and the response like this
```
[
{
id: 2,
gym_name: "Technogym gaza تكنو جيم",
logo: "https://bit.ly/3vrJEGt",
email: "techno.gym@gmail.com",
phone: "082621439",
city: "غزة",
description: "نادي رياضي يحتوي على ملعب صالة لكمال الاجسام ,بالضافة بها مسبح . أوقات الدوام من 8-12 صباحا ومن 12-16 مساء",
features: [
"مسبح",
"ميدان تنافسي",
"ملعب رياضي"
],
images: [
{
pathUrl: "https://bit.ly/3y0jWdK"
}
]
}
]
```

### CODE REVIEW

**Review by AhmedQeshta — APPROVED** [review]

File: server/controllers/gyms/SerachGymByName.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import CustomError from '../../utils';
+
+export default async function searchGymByName(req: Request, res: Response, next: NextFunction) {
+  try {
+    const gyms = await Gym.findAll({
+      where: {
+        gym_name: {
+          [Sequelize.Op.like]: `%${req.query.q}%`,
+        },
+      },
```
Comment (LinaYahya) [inline server/controllers/gyms/SerachGymByName.ts:14]: try to exclude passwords from gym data,
And exclude other fields you don't need,
Also, I guess we need a Name for the user who owns the review

File: server/controllers/gyms/SerachGymByName.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        {
+          model: Image,
+        },
+        {
+          model: Review,
+        },
+      ],
+    });
+    if (gyms.length === 0) {
+      throw new CustomError('No gyms found', 404);
```
Comment (LinaYahya) [inline server/controllers/gyms/SerachGymByName.ts:25]:
```suggestion
      throw new CustomError('No gyms found', 404); // this is not a server error, Just send a response with empty array results

```

**Review by LinaYahya — COMMENTED** [review]
What if the user searches about `techno` in lower case,
Try to find a way to ignore these differences

**Review by LinaYahya — APPROVED** [review]

**Review by rawaneyad — APPROVED** [review]
Great Job 👍🏼

**Pull request merged**